src/OpenTK.Mathematics/Vector/Vector4h.cs

```diff
@@ -1144,25 +1150,28 @@ public Vector4d ToVector4d()
         }
 
         /// <summary>
-        /// Converts OpenTK.Vector4 to OpenTK.Half4.
+        /// Returns a pointer to the first element of the specified instance.
         /// </summary>
-        /// <param name="v4f">The Vector4 to convert.</param>
-        /// <returns>The resulting Half vector.</returns>
+        /// <param name="v">The instance.</param>
+        /// <returns>A pointer to the first element of v.</returns>
         [Pure]
-        public static explicit operator Vector4h(Vector4 v4f)
+        public static unsafe explicit operator Half*(Vector4h v)
         {
-            return new Vector4h(v4f);
+            return &v.X;
         }
 
         /// <summary>
-        /// Converts OpenTK.Vector4d to OpenTK.Half4.
+        /// Returns a pointer to the first element of the specified instance.
         /// </summary>
-        /// <param name="v4d">The Vector4d to convert.</param>
-        /// <returns>The resulting Half vector.</returns>
+        /// <param name="v">The instance.</param>
+        /// <returns>A pointer to the first element of v.</returns>
         [Pure]
-        public static explicit operator Vector4h(Vector4d v4d)
+        public static explicit operator IntPtr(Vector4h v)
         {
-            return new Vector4h(v4d);
+            unsafe
+            {
+                return (IntPtr)(&v.X);
+            }
```

Comment:
What is going on here? Why are we removing these casts and adding casts to `Half*` and `IntPtr`, it makes no sense.

Reply:
These two casts were duplicates that are already present in `Vector4` and `Vector4d`, so I removed them from here.
I also added missing casts to IntPtr and T* to match other vector classes.

Reply:
See:
https://github.com/opentk/opentk/blob/ce54c2cf088798c92eb6e75e9c1be538732080d8/src/OpenTK.Mathematics/Vector/Vector4.cs#L2010
https://github.com/opentk/opentk/blob/ce54c2cf088798c92eb6e75e9c1be538732080d8/src/OpenTK.Mathematics/Vector/Vector4d.cs#L2005

Reply:
Ok fair enough, but then I think that we should remove the `T*` and `IntPtr` casts as it's better and less error prone to take the address of the thing yourself `&v.X`.
Because the casts can easily cause problems, e.g. `(float*)new Vector3(1f,2f,3f)`. This will give you an invalid pointer.

Reply:
It's indeed dangerous for peoples that are not familiar with C/C++ and unmanaged memory but as these vectors are meant to be passed to low-level graphics API, I think these are good to keep.

I myself use these casts in my ShaderPipeline class:
```cs
public unsafe void SetValue(string variableName, params Vector3[] data)
{
    fixed (Vector3* ptr = data)
    {
        GL.ProgramUniform3(_handle, GetLocation(variableName), data.Length, (float*)ptr);
    }
}
```

Reply:
Now that I think about it, we could replace these casts with methods containing `Unsafe` in their name to clearly state "Don't use it if you don't know what you are doing"

Reply:
You can't use the `T*` cast without unsafe. `IntPtr` barely has any usefulness when you can have typed pointers.
We should just remove these casts, they are just wasted code and they make people more likely to mess up their code.

Reply:
> This code sample is not using these casts. You already have a pointer (`Vector3*`), which you can cast to any other pointer. These casts are never used and are just error prone.

You're right, im not even using these casts actually haha.
If we can already do what I do without these dangerous casts, yeah there is no point in keeping them.